providers/framework.rb

```diff
@@ -1,131 +0,0 @@
-#
-# Cookbook Name:: ms_dotnet
-# Provider:: framework
-# Author:: Baptiste Courtois (<b.courtois@criteo.com>)
-#
-# Copyright (C) 2015-2016, Criteo.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-# http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-use_inline_resources
-# The provides method is available on chef >= 12.0 only
-provides :ms_dotnet_framework, os: 'windows' if respond_to?(:provides)
-
-def whyrun_supported?
```

Comment:
You are removing whyrun_support. Please add it to the resource.
I remember having to use the below code to properly support `whyrun` in chef 12.XX:
```ruby
action_class do
  def whyrun_supported?
    true
  end
end
```